### ISSUE TITLE
Order ID not present in order details after successful transaction with the delayed notification payment method

### ISSUE THREAD

**Issue by timur27** [Automattic/woocommerce-payments#6142]
### Describe the bug
Delayed payment methods are those where the funds are only available after some time. In live mode, such payments might take up to 14 business days. This behavior is also implemented and can be tested in the testing environment, but the waiting time is 3 minutes. Sofort payment method is one of these types of payments. After authorizing the payment, it's Pending and updates to the Successful state after 3 minutes. The problem this issue needs to resolve is that for Sofort, the Order ID isn't visible in the order details after the payment is made. It's provided only after 3 minutes when the status changes to _Processing_ in WC Orders.

### To Reproduce
1. Enable split UPE
2. Checkout using Sofort on the shortcode checkout and authorize the payment
3. Go to the Order details in the admin panel, and right below _Order #XYZ details_ title, click on the hyperlinked payment intent which will redirect to the Payment details page
4. On the Payment details page, confirm there is no Order ID in the _Order section
5. Wait 3 minutes and refresh, confirm that the order ID is present

### Actual behavior
Order ID appears only after the Order status changes to _Processing_. 

### Screenshots
![Sofort PI with missing order details](https://user-images.githubusercontent.com/22319444/234231464-b9fa6f3a-0f76-40b7-a536-0df1fbaf5078.png)
_Sofort pending transaction with missing order details_

![Sofort PI has order details when paid](https://cdn-std.droplr.net/files/acc_1104206/npFUBN)
_Sofort transaction has order details after payment is made_

### Expected behavior
Order ID appears right after the Order has been placed and is in _Pending_ status. This is the expected behavior for the split UPE and the deferred intent creation UPE.

**Comment by timur27** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Heisenberg-63586f2ec60359001310fa6a/issues/Automattic/woocommerce-payments/6142?planning-poker) with Zenhub @FangedParakeet @mdmoore

**Comment by FangedParakeet** [comment]
I believe this has been indirectly fixed by #6170 and can confirm that I can no longer repeat this on the `develop` branch with either the original split-UPE or new deferred intent UPE flags enabled. Hereby happily closing out this issue. 🎉
